### PR TITLE
Import bitcoin neon controller

### PR DESCRIPTION
**Pull request by lgalabru**
This PR aims at importing the neon-bitcoin-controller, orchestrating the bitcoind instance supporting neon, in the stacks-blockchain workspace.
I ll delete the private repo once this PR is merged.

### CODE REVIEW

File: testnet/bitcoin-neon-controller/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM rust as builder
+WORKDIR /src/bitcoin-neon-controller
+COPY . .
+RUN cd /src/bitcoin-neon-controller && cargo build --target x86_64-unknown-linux-gnu --release
+RUN cargo install --target x86_64-unknown-linux-gnu --path /src/bitcoin-neon-controller
+
+FROM alpine
+ARG GLIBC_VERSION="2.31-r0"
```
Comment (jcnelson) [inline testnet/bitcoin-neon-controller/Dockerfile:8]: Why is the Docker file installing glibc?  The bitcoin neon controller works just fine with musl -- I've been using it for testing for the last week without issues.

File: testnet/bitcoin-neon-controller/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+/// JSONRPC Request
+pub struct RPCRequest {
+    /// The name of the RPC call
+    pub method: String,
+    /// Parameters to the RPC call
+    pub params: Vec<serde_json::Value>,
+    /// Identifier for this Request, which should appear in the response
+    pub id: serde_json::Value,
+    /// jsonrpc field, MUST be "2.0"
+    pub jsonrpc: Option<String>,
```
Comment (jcnelson) [inline testnet/bitcoin-neon-controller/src/main.rs:223]: If this `MUST be "2.0"`, should this just be a `String`?

File: testnet/bitcoin-neon-controller/Dockerfile (hunk 1 above)
Comment (lgalabru) [reply testnet/bitcoin-neon-controller/Dockerfile:8]: will defer to @wileyj, who wrote the Dockerfile

File: testnet/bitcoin-neon-controller/src/main.rs (hunk 2 above)
Comment (lgalabru) [reply testnet/bitcoin-neon-controller/src/main.rs]: Addressed with https://github.com/blockstack/stacks-blockchain/pull/1518/commits/8ff29538c69f3ba2ce121336c6ec087be3d812bc

**Comment by lgalabru** [comment]
I'm merging, and will address the glibc feedback in a separate PR after running some tests with @wileyj.

**Pull request merged**